server/api/student/student.controller.js

```diff
@@ -84,6 +87,91 @@ exports.outreachCounts = function(req, res) {
   });
 };
 
+/**
+ * Get summary of outreaches.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get outreach summary for assignment school
+ * - manager+ will get outreach summary for all schools
+ */
+exports.outreachSummary = function(req, res) {
+  var pipeline = [{
+   $group: {
+     _id: {student: '$student', type: '$type'},
+     count: {$sum: 1}
+   }
+  }, {
+   $group: {
+     _id: '$_id.student',
+     counts: {$push: {type: '$_id.type', count: '$count'}}
+   }
+  }, {
+    $project: {
+      _id: 0,
+      student: '$_id',
+      counts: 1
+    }
+  }];
+  if (req.user.role === 'teacher') {
+    pipeline.unshift({
+       $match: {school: req.user.assignment}
+    });
+  }
+  Outreach.aggregate(pipeline, function(err, results) {
+    if (err) handleError(res, err);
+    Student.populate(results, {path: 'student'})
+      .then(function(withStudent) {
+        School.populate(withStudent, 
+          {path: 'student.currentSchool'})
+            .then(function(final) {
+              return res.status(200).json(final);
+            });
+      });
+  });
+};
+
+/**
+ * Get summary of interventions.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get intervention summary for assignment school
+ * - manager+ will get intervention summary for all schools
+ */
+exports.interventionSummary = function(req, res) {
+  var pipeline = [{
+    $group: {
+      _id: {student: '$student', school: '$school' },
+      records: { $push: '$$ROOT' }
+    } 
+  }, {
+    $project: {
+      _id: 0,
+      student: '$_id.student',
+      school: '$_id.school',
+      // Might need to exclude fields if implemented
+      records: 1
+    }
+  }];
+  if(req.user.role === 'teacher') {
+    pipeline.unshift({ 
+      $match: { school: req.user.assignment } 
+    });
+  }
+  Intervention.aggregate(pipeline, function(err, results) {
+    if (err) handleError(res, err);
+    Student.populate(results, {path: 'student'})
```

Comment:
We should use `select` options for this populate and the school populate. This will greatly reduce the size of the returned object.